com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/archive/publisher/ZipPublisher.java

```diff
@@ -41,40 +41,46 @@ public ZipPublisher(PublisherSettings settings) {
-	        } catch (IOException exception) {
-	        	throw new CompileExceptionError("Unable to write to zip archive for liveupdate resources: " + exception.getMessage(), exception);
-	        } finally {
-	            IOUtils.closeQuietly(zipOutputStream);
-	        }
-	        
-	        File exportFilehandle = new File(this.getPublisherSettings().getZipFilepath(), this.resourcePackZip.getName());
-	        Files.move(this.resourcePackZip.toPath(), exportFilehandle.toPath(), StandardCopyOption.REPLACE_EXISTING, StandardCopyOption.ATOMIC_MOVE);
-    	} catch (IOException exception) {
-        	throw new CompileExceptionError("Unable to create zip archive for liveupdate resources: " + exception.getMessage(), exception);
-    	}
+        try {
+            String tempFilePrefix = "defold.resourcepack_" + this.platform + "_";
+            this.resourcePackZip = File.createTempFile(tempFilePrefix, ".zip");
+            FileOutputStream resourcePackOutputStream = new FileOutputStream(this.resourcePackZip);
+            ZipOutputStream zipOutputStream = new ZipOutputStream(resourcePackOutputStream);
+            try {
+                for (String hexDigest : this.getEntries().keySet()) {
+                    File fhandle = this.getEntries().get(hexDigest);
+                    ZipEntry currentEntry = new ZipEntry(fhandle.getName());
+                    zipOutputStream.putNextEntry(currentEntry);
+
+                    FileInputStream currentInputStream = new FileInputStream(fhandle);
+                    int currentLength = 0;
+                    byte[] currentBuffer = new byte[1024];
+                    while ((currentLength = currentInputStream.read(currentBuffer)) > 0) {
+                        zipOutputStream.write(currentBuffer, 0, currentLength);
+                    }
+
+                    zipOutputStream.closeEntry();
+                    IOUtils.closeQuietly(currentInputStream);
+                }
+            } catch (FileNotFoundException exception) {
+                throw new CompileExceptionError("Unable to find required file for liveupdate resources: " + exception.getMessage(), exception);
+            } catch (IOException exception) {
+                throw new CompileExceptionError("Unable to write to zip archive for liveupdate resources: " + exception.getMessage(), exception);
+            } finally {
+                IOUtils.closeQuietly(zipOutputStream);
+            }
+
+            File exportFilehandle = new File(this.getPublisherSettings().getZipFilepath(), this.resourcePackZip.getName());
+            if (!exportFilehandle.isAbsolute())
+            {
+                File cwd = new File(System.getProperty("user.dir"));
+                exportFilehandle = new File(cwd, exportFilehandle.getPath());
+            }
+
+            Files.move(this.resourcePackZip.toPath(), exportFilehandle.toPath(), StandardCopyOption.REPLACE_EXISTING);
```

Comment:
Also removed ATOMIC_MOVE since it doesn't allow copying between different file stores.